backend/models/project.js

```diff
@@ -326,46 +264,134 @@
 		return modelsSettings;
 	};
 
-	const Project = ModelFactory.createClass(
-		"Project",
-		schema,
-		() => {
-			return "projects";
-		}
-	);
+	Project.isProjectAdmin = async function(teamspace, model, user) {
+		const projection = { "permissions": { "$elemMatch": { user: user } }};
+		const project = await Project.findOneProject(teamspace, {models: model}, projection);
+		const hasProjectPermissions = project && project.permissions.length > 0;
 
-	Project.updateProject = async function(account, projectName, data) {
-		const project = await this.findOne({ account }, { name: projectName });
+		return hasProjectPermissions && project.permissions[0].permissions.includes(C.PERM_PROJECT_ADMIN);
+	};
+
+	Project.removeProjectModel = async function(teamspace, model) {
+		return db.update(teamspace, PROJECTS_COLLECTION_NAME, { models: model }, { "$pull" : { "models": model}}, {"multi": true});
+	};
+
+	Project.setUserAsProjectAdmin = async function(teamspace, project, user) {
+		const projectObj = await Project.findOneProject(teamspace, {name: project});
+		const projectPermission = { user, permissions: ["admin_project"]};
+
+		return await Project.updateAttrs(teamspace, project, { permissions: projectObj.permissions.concat(projectPermission) });
+	};
+
+	Project.setUserAsProjectAdminById = async function(teamspace, project, user) {
```

Comment:
I think we should refactor this and 279 to call a private function `setUsersAsProjectByQuery` which both calls with the query, as that's the only difference between them